tests/test_storage.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+"""
+This is a unit test for representing storage in the MESSAGEix model, and
+testing the functionality of storage equations. The workflow is as follows:
+    - building a stylized MESSAGEix model
+    - adding seasonality and modifying parameters accordingly
+    - adding storage implementation
+    - testing storage functionality and equations
+
+"""
+
+from message_ix import Scenario
+
+# A flag to show if storage variables/parameters are added to ixmp java backend
+storage_in_java = False
+
+
+# A function for generating a simple MESSAGEix model
+def model_setup(scen, years):
+    scen.add_set('node', 'node')
+    scen.add_set('commodity', 'electr')
+    scen.add_set('level', 'level')
+    scen.add_set('year', years)
+    scen.add_set('type_year', years)
+    scen.add_set('technology', ['tec1', 'tec2'])
+    scen.add_set('mode', 'mode')
+    output_specs = ['node', 'electr', 'level', 'year', 'year']
+    dict_var_cost = {'tec1': 1, 'tec2': 2}
+
+    for yr in years:
+        scen.add_par('demand', ['node', 'electr', 'level', yr, 'year'],
+                     1, 'GWa')
+        for t in dict_var_cost.keys():
+            tec_specs = ['node', t, yr, yr, 'mode']
+            scen.add_par('output', tec_specs + output_specs, 1, 'GWa')
+            scen.add_par('var_cost',
+                         tec_specs + ['year'], dict_var_cost[t], 'USD/GWa')
+
+
+# A function for adding sub-annual time steps to a MESSAGEix model
+def add_seasonality(scen, seasons_dict):
+    scen.add_set('time', list(set(seasons_dict.keys())))
+    scen.add_set('lvl_temporal', 'season')
+    for h in seasons_dict.keys():
+        scen.add_set('map_temporal_hierarchy', ['season', h, 'year'])
+        scen.add_par('duration_time', [h], seasons_dict[h], '-')
+
+
+# A function for modifying model parameters after adding sub-annual time steps
+def year_to_time(scen, parname, dict_par, filters=None):
```

Comment:
can you explain a bit more what this function is doing/why it is needed?

also, it looks like the `filters` argument is not used - can it be removed?

and finally, could you please format standard docstrings? so this would be:

```
def year_to_time(scen, parname, dict_par):
 """A function for modifying model parameters after adding sub-annual time steps (but now adding more explanation)"""
```

Reply:
@Matt! This function takes a MESSAGE parameter with the time index of 'year' and modifies the parameters for new "time" indexes, let's say months. It modifies the 'value' if needed too, based on the sahre of each time step from the yearly values. _filters_ could be used to load only a part of a parameter not all the table. As you noticed it's not used in this test and I removed it. Thanks!